code/modules/tools/tool_extension_variable.dm

```diff
@@ -0,0 +1,37 @@
+/datum/extension/tool/variable
+	var/current_tool
+	var/list/tool_config_sounds // Associative list of tool to sound/list of sounds used to override the archetype use sounds.
+
+/datum/extension/tool/variable/New(datum/holder, list/_tool_values, list/_tool_use_sounds, list/_tool_config_sounds)
+	. = ..()
```

Comment:
I recall there was a post by Lummux, stating that `New/()` should never return anything (at by extension have its `.` value adjusted)